src/utils/test/packet_collectors.rs

```diff
@@ -33,3 +34,39 @@ impl<T: Debug> Future for ExhaustiveDrain<T> {
         }
     }
 }
+
+/// Exhaustive Collector works like Exhaustive Drain in that it continously polls for packets until it
+/// receives a None, but it also will write that value out to the provided channel, so that the packet
+/// may be compared in a test.
+pub struct ExhaustiveCollector<T: Debug> {
+    id: usize,
+    stream: ElementStream<T>,
+    packet_dump: crossbeam_channel::Sender<T>
+}
+
+impl<T: Debug> ExhaustiveCollector<T> {
+    pub fn new(id: usize, stream: ElementStream<T>, packet_dump: crossbeam_channel::Sender<T>) -> Self {
+        ExhaustiveCollector { id, stream, packet_dump }
+    }
+}
+
+impl<T: Debug> Future for ExhaustiveCollector<T> {
```

Comment:
I wish there was a way to reduce redundancy, since `ExhaustiveDrain` and `ExhaustiveCollector` are so similar.

Could `ExhaustiveDrain` just contain an `ExhaustiveCollector` that does nothing with its channel?

Reply:
Let's leave this for a separate PR.  We are going to drop many tests all together.